BentoKit/BentoKit/Screen/ScreenLifecycleEvent.swift

```diff
@@ -0,0 +1,35 @@
+public protocol ScreenLifecycleObserving {
+    func send(_ event: ScreenLifecycleEvent)
+}
+
+public struct ScreenLifecycleEvent {
+    public let status: Status
+    public let isBeingPresentedInitially: Bool
```

Comment:
I guess this would not be necessary if we have `viewDidLoad`.

Reply:
Having `didLoad` does not change the fact that there are at least 8 different scenarios:

* will/did appear AND (is moving to parent OR is being presented)
* will/did disappear AND (is moving from parent OR is being dismissed)
* will/did appear AND NEITHER (is moving to parent OR is being presented)
* will/did disappear AND NEITHER (is moving from parent OR is being dismissed)

`viewDidLoad` may also happen way before `viewWillAppear`, so it might not necessarily be what one wants.

Reply:
> and a view controller might be reused
Well I do not think it's a good practice. ¯\_(ツ)_/¯